### PR TITLE
Random Theme Updates

### PR DESCRIPTION
**Pull request by huayunh**
The previous PR on themes was done in a hurry and there are still a bunch of things. 

#### Changes proposed in this Pull Request
- Cleaned up folder structure of the themes folder
- Added a witch for the Halloween theme
- Added two themes for Christmas and ... 

#### Screenshots
image

### CODE REVIEW

File: src/__configuration__/themes/default.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                    color: Colors.white[50],
+                    backgroundColor: Colors.blue[700],
+                    '& .MuiInputBase-root': {
+                        color: Colors.white[50],
+                    },
+                    '& .MuiSelect-icon': {
+                        color: Colors.white[50],
+                    },
+                },
+            },
```
Comment (joebochill) [inline src/__configuration__/themes/default.ts:31]: Why are we overriding our default (PX Blue) theme for our default theme?

File: src/__configuration__/themes/halloween.ts [hunk 2]
```diff
@@ -71,3 +77,23 @@ export const HalloweenTheme = {
         },
     },
 };
+
+export const HalloweenSchedule: Schedule = {
+    start: new Date(0, 9, 14), // Oct 14
+    end: new Date(0, 10, 13), // Nov 13
```
Comment (joebochill) [inline src/__configuration__/themes/halloween.ts:83]: 2 weeks after halloween seems like a lot. Maybe 1?

File: src/__configuration__/themes/christmas.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import * as PXBThemes from '@pxblue/react-themes';
+import * as Colors from '@pxblue/colors';
+import * as BrandingColors from '@pxblue/colors-branding';
+import ChristmasEve from '../../app/assets/themes/christmas-eve.png';
+import AppBarTile from '../../app/assets/themes/christmas-appbar-tile.png';
+import { Schedule } from './types';
+
+export const ChristmasSchedule: Schedule = {
```
Comment (joebochill) [inline src/__configuration__/themes/christmas.ts:8]: Hate to be this guy, but if we're going to do a "Christmas" theme instead of a generic 'holiday' or 'winter' theme, then we're probably also going to need a Kwanzaa theme and a Hanukkah theme.

File: src/__configuration__/themes/christmas.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    start: new Date(0, 11, 5), // Dec 05
+    end: new Date(0, 11, 26), // Dec 26
```
Comment (joebochill) [inline src/__configuration__/themes/christmas.ts:10]: Suggest we keep this just to the two or three days around christmas, and show the Hanukkah/Kwanzaa themes during their respective periods (we'l need to add a note to update the config since these dates change each year). Then maybe a generic winter theme for the remaining days (if there are any).

File: src/__configuration__/themes/halloween.ts [hunk 5]
```diff
@@ -71,3 +77,23 @@ export const HalloweenTheme = {
...
+    config: {
+        theme: HalloweenTheme,
+        landingPageBanner: {
+            src: castles,
+            backgroundSize: 'cover',
+            backgroundPosition: 'bottom',
+            backgroundRepeat: 'no-repeat',
+            cursor: `url("${witch}"), auto`,
```
Comment (joebochill) [inline src/__configuration__/themes/halloween.ts:91]: This is cute

File: src/__configuration__/themes/schedule.ts [hunk 6]
```diff
@@ -1,87 +1,15 @@
...
+import { ChristmasSchedule } from './christmas';
+import { defaultTheme } from './default';
+import { Schedule, SiteConfig } from './types';
+
+// Add more holiday themes here
+export const schedule: Schedule[] = [ChristmasSchedule, HalloweenSchedule, MayTheForthSchedule];
+
+/**
+ * @returns a site configuration given the currently applied holiday theme
+ */
```
Comment (joebochill) [inline src/__configuration__/themes/schedule.ts:12]: Might be nice to see a comment somewhere that shows the schedule in plain english, e.g.:
```
/**
* ...
* Dec 23 - Dec 26: Christmas
* Dec 29 - Jan 2: New Years
**/
```

File: src/index.css [hunk 7]
```diff
@@ -113,3 +113,8 @@ p.MuiTypography-h5.MuiTypography-paragraph {
 .halloween.ph-item .ph-row > div > * {
     background-color: #2b353a;
 }
+
+/* May the Forth Theme */
+.may-the-forth {
+    cursor: url('./app/assets/themes/vulcan.png'), auto;
```
Comment (joebochill) [inline src/index.css:119]: This cursor has some jankiness when you're moving around the permanent drawer header. I prefer what you did for the witch (just on the banner).

Also, don't let Kyle see this or he will flip out. Vulcan == Star Trek, not Star Wars

**Review by joebochill — COMMENTED** [review]
Can we get a tree in the Christmas tile?

File: src/__configuration__/themes/christmas.ts (hunk 3 above)
Comment (joebochill) [reply src/__configuration__/themes/christmas.ts:8]: If there is time, I'm sure our counterparts in Pune would be happy to see a Diwali theme in mid November too

Comment (joebochill) [reply src/__configuration__/themes/christmas.ts:8]: We don't have to do all of these right now though. But we should have some backlog items to do them in time for their respective holidays.

File: src/index.css (hunk 7 above)
Comment (huayunh) [reply src/index.css:119]: oooof, should've looked more carefully next time I check these up

File: src/__configuration__/themes/default.ts (hunk 1 above)
Comment (huayunh) [reply src/__configuration__/themes/default.ts:31]: Because it will otherwise be a black500 on blue500 for resource page & roadmap page

File: src/__configuration__/themes/halloween.ts (hunk 2 above)
Comment (huayunh) [reply src/__configuration__/themes/halloween.ts:83]: Changing to Nov 2... Don't feel like doing more Halloween things after the Election day

File: src/index.css (hunk 7 above)
Comment (huayunh) [reply src/index.css:119]: > has some jankiness when you're moving around the permanent drawer header.

Sorry what kinds of the "jankiness" are you thinking here? I tried on my end but it seems fine...

Comment (huayunh) [reply src/index.css:119]: How is this guy? 

image

File: src/__configuration__/themes/christmas.ts (hunk 3 above)
Comment (huayunh) [reply src/__configuration__/themes/christmas.ts:8]: >  I'm sure our counterparts in Pune would be happy to see a Diwali theme in mid November too

Eeeek, I want to just pay someone to do us a banner illustration for that

Comment (huayunh) [reply src/__configuration__/themes/christmas.ts:8]: > a Kwanzaa theme and a Hanukkah theme.

I... am not even aware of these holidays. 🤦‍♀️  They will go to the backlog for sure, and let someone who knows the culture better handle it lest I offend anyone with that Vulcan / Star Trek thing.

File: src/index.css (hunk 7 above)
Comment (huayunh) [reply src/index.css:119]: I didn't want to do make it landing-page-only, because I wasn't able to come up with a proper design for the app bar tile (someone knows the culture better can do it). But then the rest of the theme looks just like a regular blue theme. This is why I want to get the to show up everywhere.

File: src/__configuration__/themes/default.ts (hunk 1 above)
Comment (joebochill) [reply src/__configuration__/themes/default.ts:31]: So, should we be making this change in the themes repo then?

File: src/__configuration__/themes/halloween.ts (hunk 2 above)
Comment (joebochill) [reply src/__configuration__/themes/halloween.ts:83]: Election Day theme!

File: src/__configuration__/themes/christmas.ts (hunk 3 above)
Comment (joebochill) [reply src/__configuration__/themes/christmas.ts:8]: Sounds good, as long as we have a story for them we should be good

File: src/index.css (hunk 7 above)
Comment (joebochill) [reply src/index.css]: Jankiness was around the cursor showing up unexpectedly when moving over the drawer header. I'll send you a screen recording to try to illustrate.

Oh, I have some thoughts for the tile: Darth Vader mask, Stormstrooper, AT-AT
image
image
image

File: src/__configuration__/themes/christmas.ts (hunk 3 above)
Comment (huayunh) [reply src/__configuration__/themes/christmas.ts:8]: PXBLUE-1674, listed as priority items to groom for next sprint.

File: src/__configuration__/themes/default.ts (hunk 1 above)
Comment (huayunh) [reply src/__configuration__/themes/default.ts:31]: That is correct. 

I was thinking that this will happen when we do all the dark theme updates.

Comment (joebochill) [reply src/__configuration__/themes/default.ts:31]: Can you maybe add a comment to this effect in whatever story we have on the backlog for dark theme updates?

**Review by joebochill — APPROVED** [review]
Looks awesome. One thought...could we do the star wars primary toolbar in dark gray with white (or yellow) tile svg? Or is that too much of a mix of light/dark theming?

I'm thinking this sort of vibe:
image

Like this, but not quite as opaque:
image


Edit: If you could add the 3d perspective to the tile like in the star wars text, that would be dope 🔥 , but that's asking a lot.

**Comment by huayunh** [comment]
You mean, you want dark theme for that?

**Comment by huayunh** [comment]
For that I would just change everything to dark theme, and with yellow as a primary color.
